api/v1alpha1/wildflyserver_types.go

```diff
@@ -134,8 +134,10 @@ type PodStatus struct {
 	Name  string `json:"name"`
 	PodIP string `json:"podIP"`
 	// Represent the state of the Pod, it is used especially during scale down.
-	// +kubebuilder:validation:Enum=ACTIVE;SCALING_DOWN_RECOVERY_INVESTIGATION;SCALING_DOWN_RECOVERY_DIRTY;SCALING_DOWN_CLEAN
+	// +kubebuilder:validation:Enum=ACTIVE;SCALING_DOWN_RECOVERY_INVESTIGATION;SCALING_DOWN_RECOVERY_PROCESSING;SCALING_DOWN_RECOVERY_HEURISTICS;SCALING_DOWN_CLEAN
 	State string `json:"state"`
+	// Counts the recovery attempts when there are in-doubt/heuristic transactions
+	RecoveryCounter int32 `json:"recoveryCounter"`
```

Comment:
```suggestion
	RecoveryCounter int32 `json:"recoveryCounter",omitempty`
```

Dont we need this. To avoid situation in https://issues.redhat.com/browse/JBEAP-23493? @yersan

Reply:
+1 should be: 

```
RecoveryCounter int32 `json:"recoveryCounter,omitempty"`
```

Reply:
Thanks @mchoma :-) Updated